app/controllers/users/totp_setup_controller.rb

```diff
@@ -0,0 +1,46 @@
+module Users
+  class TotpSetupController < ApplicationController
+    before_action :confirm_two_factor_authenticated
+
+    def new
+      if session[:otp_secret_key].nil?
+        session[:otp_secret_key] = current_user.generate_totp_secret
+      end
+      @qrcode = UserDecorator.new(current_user).qrcode(session[:otp_secret_key])
+    end
+
+    def confirm
+      if valid_code?
+        current_user.save!
+        process_valid_code
+      else
+        process_invalid_code
+      end
+    end
+
+    def disable
+      if current_user.otp_secret_key.present?
+        current_user.update(otp_secret_key: nil)
+      end
+      flash[:success] = t('upaya.notices.totp_disabled')
+      redirect_to edit_user_registration_path
+    end
```

Comment:
Should we have some kind of warning/confirmation to make sure the user really wants to disable TOTP?